pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -84,6 +85,17 @@ def test_categorical_preserves_tz(self):
                 result = DatetimeIndex(obj)
                 tm.assert_index_equal(result, dti)
 
+    def test_categorical_preserves_freq(self):
+        # GH33830 freq retention in categorical
+        dti = date_range("2016-01-01", periods=5)
+
+        expected = dti.freq
+
+        cat = pd.Categorical(dti)
+        result = cat.categories.freq
+
+        assert_equal(expected, result)
```

Comment:
1 . This test should go in the `pandas/tests/arrays/categorical/` directory.

2. Could you also add separate tests for the `factorize` and `MultiIndex` calls in the original issue?

3. You can just use `assert result == expected`

Reply:
@mroeschke Went ahead and moved the test to the new directory with the change you suggested. 

I purposefully didn't add tests for `factorize` and `MultiIndex` because it seems they were added in PR #38120 and still exist in the latest version.